portality/dao.py

```diff
@@ -120,19 +122,29 @@ def save(self, retries=0, back_off_factor=1, differentiate=False, blocking=False
         while attempt <= retries:
             try:
                 r = requests.post(url, data=d)
-                if 400 <= r.status_code < 500:
-                    # bad request, no retry
-                    print "bad request", r.json()
-                    traceback.print_stack(file=sys.stdout)
-                    break
-                elif r.status_code >= 500:
-                    print "server error", r.json()
-                    attempt += 1
+                if r.status_code > 400:
+                    raise ElasticSearchWriteException(u"Error on ES save. Response code {0}".format(r.status_code))
                 else:
                     break  # everything is OK, so r should now be assigned to the result
 
             except requests.exceptions.ConnectionError:
+                app.logger.exception(u"Failed to connect to ES")
                 attempt += 1
+            except ElasticSearchWriteException:
+                try:
+                    error_details = r.json()
+                except (ValueError, AttributeError):
+                    error_details = None
```

Comment:
In this case maybe use `r.text`? http://docs.python-requests.org/en/master/user/quickstart/#response-content